### PR TITLE
docs(ReadMe): Updated with correct addresses and user guide on new implementation

### LINKED ISSUE #624: [EVM -> DFC] Change TX fees variable to 0
**What?**
We will be removing the original 0.1% tx fees for users bridging from EVM -> DFC on quantum. Instead of removing the variable, we will just change the value of it to 0.

**Why?**
This decision was made to incentivise more liquidity inflows into DFC with 0% fees to enter the ecosystem. 

No change to tx fees for users bridging from DFC -> EVM.

### PR DESCRIPTION
**Pull request by Abhishekkochar**
#### What this PR does / why we need it:

- Updated read me with correct address version 5.
- Time lock contract role info

#### Which issue(s) does this PR fixes?:

Fixes #624 

#### Developer Checklist:

- [ ] Read your code changes at least once
- [ ] No console errors on web
- [ ] Tested on Light mode and Dark mode\*
- [ ] Your UI implementation visually matched the rendered design\*
- [ ] Unit tests\*
- [ ] Added e2e tests\*
- [ ] Added translations\*

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -39,54 +39,77 @@ TODO
...
+
+Gnosis safe will be implemented with both proposers and executors roles. Gnosis Safe has the authority to grant roles to addresses, as long as it holds the corresponding role itself. When revoking privileges, either the revokeRole() or renounceRole() functions must be used.
+
+To execute only Admin transactions, the developer will need to follow these steps:
+
+- First, create a transaction using Safe [guide](<(https://help.safe.global/en/articles/3738081-contract-interactions),>).
+- After providing the contract address and ABI, the developer can select the contract method (in this case, we will try to change the transaction fee).
+- Select "Schedule". According to the Schedule() function, the following arguments need to be provided: address target, uint256 value, bytes calldata data, bytes32 predecessor, bytes32 salt, and uint256 delay.
+- The target address is the Proxy Bridge. The value will usually be 0. The data is the encoded data of the function and arguments (BridgeV1Interface.encodeFunctionData('changeTxFee', [100])). The predecessor is ZERO_BYTES32, and the salt will be in incremented order (e.g., 0x0...1, 0x0...2, 0x0...3, and so on).
```
Comment (weiyuan95) [inline README.md:53]: Let's mention that predecessor should almost always be 0x0 unless we have a prerequisite transaction, and that the delay represents the minimum amount of time before the tx can be executed.

Is the delay in seconds or ms?

Comment (Abhishekkochar) [reply README.md:53]: Always in seconds. I will mentions this as well.

File: README.md [hunk 2]
```diff
@@ -35,54 +35,82 @@ TODO
 
 To change the state of any smart contract, users need to approve the smart contract of the respective token via the `approve()` function first. Once approved, user will be able to bridge the token over to DefiChain.
 
+We are implementing a TimeLock contract that will work as an Admin address for ADMIN ONLY tx. There will be 3 days delay on every operational tx except when calling `Withdraw()` function.
```
Comment (cuongquangnam) [inline README.md:38]: Timelock cannot withdraw iirc?

File: README.md [hunk 3]
```diff
@@ -35,54 +35,82 @@ TODO
...
+
+### TimeLock Contract Account Permission
+
+Gnosis safe will be implemented with both proposers and executors roles. Gnosis Safe has the authority to grant roles to addresses, as long as it holds the corresponding role itself. When revoking privileges, either the revokeRole() or renounceRole() functions must be used.
```
Comment (cuongquangnam) [inline README.md:42]: What do you mean by the corresponding role itself?

File: README.md [hunk 4]
```diff
@@ -35,54 +35,82 @@ TODO
...
+
+To execute only Admin transactions, the developer will need to follow these steps:
+
+- First, create a transaction using Safe [guide](<(https://help.safe.global/en/articles/3738081-contract-interactions),>).
+- After providing the contract address and ABI, the developer can select the contract method (in this case, we will try to change the transaction fee).
```
Comment (cuongquangnam) [inline README.md:47]: should we say clearly we should provide the TimelockController address and its abi here?

File: README.md [hunk 5]
```diff
@@ -35,54 +35,82 @@ TODO
...
+
+The reason behind choosing different salt is to avoid having the same transaction again.
+
+After scheduling the transaction using a timelock, the developer must call the execute() function with the provided arguments(same as above). If the execute() function is called before the specified `delay` time has passed, the transaction will revert with the error message "TimelockController: operation is not ready". This is because the timelock is designed to ensure that the specified delay time has elapsed before the transaction can be executed. Once the delay time has passed, the transaction can be executed normally.
+
+### Bridge Contract Account Permission
+
+There are only two roles: DEFAULT_ADMIN_ROLE and WITHDRAW_ROLE.
+
+The TimeLock contract is assigned the DEFAULT_ADMIN_ROLE, while another Gnosis Safe is assigned the WITHDRAW_ROLE. The DEFAULT_ADMIN_ROLE has the ability to change both roles to other addresses, but this change will only take effect after three days.
```
Comment (cuongquangnam) [inline README.md:62]:
```suggestion
The TimeLock contract is assigned the DEFAULT_ADMIN_ROLE, while another Gnosis Safe is assigned the WITHDRAW_ROLE. The DEFAULT_ADMIN_ROLE has the ability to grant both roles to other addresses, but this change will only take effect after three days.
```

File: README.md [hunk 6]
```diff
@@ -35,54 +35,82 @@ TODO
...
+Finally, both addresses can renounce their own roles by calling the renounceRole() function.
+
 ### Bridge ERC20 tokens - to transfer ERC20 tokens from an EOA to the Bridge
 
 Once approved, user will call the `bridgeToDeFiChain()` function with following arguments: `_defiAddress`- address on Defi Chain that receiving funds, `_tokenAddress` - ERC20 token's address and `_amount` amount to bridge over to Defi chain.
 
 ### Add supported token
 
-Only addresses with the Admin and Operational roles can call the `addSupportedTokens()` function. This sets the `_tokenCap` for an ERC20 token and ETH identified by its `_tokenAddress`. All added tokens will be instantly supported by the bridge.
+Only address with the Admin role can call the `addSupportedTokens()` function. This sets the `_tokenCap` for an ERC20 token and ETH identified by its `_tokenAddress`. All added tokens will be instantly supported by the bridge.
```
Comment (cuongquangnam) [inline README.md:72]:
```suggestion
Only address with the Admin role of the BridgeProxy contract can call the `addSupportedTokens()` function. This sets the `_tokenCap` for an ERC20 token and ETH identified by its `_tokenAddress`. All added tokens will be instantly supported by the bridge.
```

File: README.md [hunk 7]
```diff
@@ -35,54 +35,82 @@ TODO
...
 
 In case of ETH, address(0) will be used as an address.
 
 `_tokenCap` represent the maximum balance of tokens the contract can hold per `_tokenAddress`
 
 ### Remove supported token
 
-Only addresses with the Admin and Operational role can call the `removeSupportedTokens()` function. This also sets the `_tokenCap` to `0`.
+Only address with the Admin role can call the `removeSupportedTokens()` function. This also sets the `_tokenCap` to `0`.
```
Comment (cuongquangnam) [inline README.md:80]:
```suggestion
Only address with the Admin role of BridgeProxy contract can call the `removeSupportedTokens()` function. This also sets the `_tokenCap` to `0`.
```

File: README.md [hunk 8]
```diff
@@ -35,54 +35,82 @@ TODO
...
 
 ### Withdraw
 
-`withdraw()` function when called will withdraw an ERC20 token and ETH (address == 0x0). Only the address with the Admin role can call this function.
+`withdraw()` function when called will withdraw an ERC20 token and ETH (address == 0x0). Only the address with the WITHDRAW role can call this function.
```
Comment (cuongquangnam) [inline README.md:84]:
```suggestion
`withdraw()` function when called will withdraw an ERC20 token and ETH (address == 0x0). Only the address with the WITHDRAW role of BridgeProxy contract can call this function.
```

File: README.md [hunk 9]
```diff
@@ -35,54 +35,82 @@ TODO
...
 
 ### FlushFund
 
 `flushFund` function to flush the excess funds `(token.balanceOf(Bridge) - tokenCap)` across supported tokens to a hardcoded address (`flushReceiveAddress`) anyone can call this function. This applies to all tokens and ETH.
```
Comment (cuongquangnam) [inline README.md:88]: should change documentation about flushFund here?

File: README.md [hunk 10]
```diff
@@ -35,54 +35,82 @@ TODO
...
 
 ### Change Flush Receive Address
 
-Both the Admin and Operational addresses can change `flushReceiveAddress`.
+Admin address can change `flushReceiveAddress`.
```
Comment (cuongquangnam) [inline README.md:92]: Admin address of BridgeProxy contract

File: README.md [hunk 11]
```diff
@@ -35,54 +35,82 @@ TODO
...
 `changeFlushReceiveAddress` function will reset the `flushReceiveAddress` to new address.
 
 ### Change relayer address
 
-Both the Admin and Operational addresses can change `relayerAddress`.
+Admin address can change `relayerAddress`.
```
Comment (cuongquangnam) [inline README.md:97]: Admin address of BridgeProxy contract

File: README.md [hunk 12]
```diff
@@ -35,54 +35,82 @@ TODO
...
 
 The relayer address will primarily be used for verifying the signature that is signed by the server. The server will need to sign with the corresponding private key of the relayer address.
 
 ### Transaction fee change
 
-Only addresses with Admin and Operational roles can change `transactionFee`.
+Only address with Admin role can change `transactionFee`.
```
Comment (cuongquangnam) [inline README.md:103]: Admin role of bridgeproxy contract

File: README.md [hunk 13]
```diff
@@ -35,54 +35,82 @@ TODO
...
 
-Initial fee will be set to 0.1%. This means that if the user bridges `X` tokens, 99.9% of X will be bridged. The other 0.1% will be taken as fees and sent to `communityWallet`.
+Initial fee will be set to 0%. This means that if the user bridges `X` tokens, 100% of X will be bridged to defiChain. If in future, fee > 0, respected amount will be sent to `communityWallet`.
 
 ### Change Tx Fee Address
 
-Only addresses with admin and Operational roles can change `communityWallet`. This is where the tx fees upon bridging will go.
+Only address with admin role can change `communityWallet`. This is where the tx fees upon bridging will go.
```
Comment (cuongquangnam) [inline README.md:109]: admin role of bridgeProxy contract

File: README.md [hunk 14]
```diff
@@ -35,54 +35,82 @@ TODO
...
 
 ### Change Token Cap
 
-Only addresses with admin and Operational roles can change `tokenCap`.
+Only address with admin role can change `tokenCap`.
```
Comment (cuongquangnam) [inline README.md:113]: admin role of bridgeproxy contract

File: README.md [hunk 15]
```diff
@@ -143,14 +175,16 @@ Admins can send ERC20 tokens via the `transfer(address _to, uint256 _amount)` fu
 
 ### Withdrawing funds
 
-ERC20 tokens can be withdrawn by the Admin address only via the `withdraw(address _tokenAddress, uint256 amount)` function.
+ETH and ERC20 tokens can be withdrawn by the address with WITHDRAW_ROLE only via the `withdraw(address _tokenAddress, uint256 amount)` function.
```
Comment (cuongquangnam) [inline README.md:178]: WITHDRAW_ROLE of bridgeproxy contract

File: README.md [hunk 16]
```diff
@@ -143,14 +175,16 @@ Admins can send ERC20 tokens via the `transfer(address _to, uint256 _amount)` fu
...
 
-## Admin and Operational addresses - Gnosis safe
+## Admin address - Gnosis safe
 
-Admin and Operational addresses will be Gnosis safes, ideally will be with at least 3 owners with a 2/3 quorum.
+Admin address will be Gnosis safe, ideally will be with at least 3 owners with a 2/3 quorum.
```
Comment (cuongquangnam) [inline README.md:182]: Admin address of TimelockController contract

Comment (cuongquangnam) [reply README.md:182]: and how about withdraw_role of BridgeProxy? 
I think let this be discussed off-Github

**Review by cuongquangnam — CHANGES_REQUESTED** [review]

File: README.md (hunk 2 above)
Comment (Abhishekkochar) [reply README.md:38]: Yes, this is what it says as well.

File: README.md (hunk 4 above)
Comment (Abhishekkochar) [reply README.md:47]: Guide is pretty explanatory. They will have to look this as a guide anyway.

File: README.md (hunk 6 above)
Comment (Abhishekkochar) [reply README.md:72]: I don't think we need to add of `of the BridgeProxy contract` as the function we are calling is of BridgeProxy contract.

File: README.md (hunk 15 above)
Comment (Abhishekkochar) [reply README.md:178]: `WITHDRAW_ROLE ` is only in BridgeProxy contract.

File: README.md (hunk 2 above)
Comment (cuongquangnam) [reply README.md:38]: hmm, the way it writes here indicates that Timelock contract can withdraw?

File: README.md (hunk 3 above)
Comment (Abhishekkochar) [reply README.md:42]: If multi-sig wallet wants to give other wallet a proposers role, they can only do so if the executing account hold that role as well.

File: README.md [hunk 17]
```diff
@@ -31,58 +31,92 @@ TODO
...
-## Operational Transactions
+####
 
 To change the state of any smart contract, users need to approve the smart contract of the respective token via the `approve()` function first. Once approved, user will be able to bridge the token over to DefiChain.
 
+We are implementing a TimeLock contract that will work as an Admin address for ADMIN ONLY tx. There will be 3 days delay on every operational tx except when calling `Withdraw()` function, TimeLock contract is not able to call this function.
+
+### TimeLock Contract Account Permission
+
+Gnosis safe will be implemented with both proposers and executors roles. Gnosis Safe has the authority to grant roles to addresses, as long as it holds the corresponding role itself. When revoking privileges, either the revokeRole() or renounceRole() functions must be used.
```
Comment (cuongquangnam) [inline README.md:42]:
```suggestion
Gnosis safe will be implemented with both proposers and executors roles. Only the Timelock smart contract has the role (TIMELOCK_ADMIN_ROLE) to grant roles for now. If we want to grant new roles to new addresses, have to go through a round of scheduling and executing the grantRole functions through the Timelock contract. When revoking privileges, either the revokeRole() or renounceRole() functions must be used.
```

File: README.md [hunk 18]
```diff
@@ -31,58 +31,92 @@ TODO
...
 TODO
 
-## Operational Transactions
+####
 
 To change the state of any smart contract, users need to approve the smart contract of the respective token via the `approve()` function first. Once approved, user will be able to bridge the token over to DefiChain.
 
+We are implementing a TimeLock contract that will work as an Admin address for ADMIN ONLY tx. There will be 3 days delay on every operational tx except when calling `Withdraw()` function, TimeLock contract is not able to call this function.
+
+### TimeLock Contract Account Permission
```
Comment (cuongquangnam) [inline README.md:40]: ## Timelock Contract Operation

File: README.md [hunk 19]
```diff
@@ -31,58 +31,92 @@ TODO
...
+
+Salt can be `0x0000000000000000000000000000000000000000000000000000000000000000` for 1st transaction,
+`0x0000000000000000000000000000000000000000000000000000000000000001` for 2nd transaction and so on.
+
+The reason behind choosing different salt is to avoid having the same transaction again.
+
+After scheduling the transaction using a timelock, the developer must call the execute() function with the provided arguments(same as above). If the execute() function is called before the specified `delay` time has passed, the transaction will revert with the error message "TimelockController: operation is not ready". This is because the timelock is designed to ensure that the specified delay time has elapsed before the transaction can be executed. Once the delay time has passed, the transaction can be executed normally.
+
+## Operational Transactions
```
Comment (cuongquangnam) [inline README.md:58]:
```suggestion
## Bridge Contract operation
```

File: README.md [hunk 20]
```diff
@@ -31,58 +31,92 @@ TODO
...
+
+### Bridge Contract Account Permission
+
+There are only two roles: DEFAULT_ADMIN_ROLE and WITHDRAW_ROLE.
+
+The TimeLock contract is assigned the DEFAULT_ADMIN_ROLE, while another Gnosis Safe is assigned the WITHDRAW_ROLE. The DEFAULT_ADMIN_ROLE has the ability to grant both roles to other addresses, but this change will only take effect after three days.
```
Comment (cuongquangnam) [inline README.md:64]: actually, from the perspective of the bridge contract, the change will happen instantly after default_admin_role call the grantRole function?

File: README.md (hunk 15 above)
Comment (cuongquangnam) [reply README.md:200]: should we remove this one, cos it is a duplicate of a part before

**Review by cuongquangnam — CHANGES_REQUESTED** [review]

File: README.md [hunk 21]
```diff
@@ -37,9 +37,9 @@ To change the state of any smart contract, users need to approve the smart contr
 
 We are implementing a TimeLock contract that will work as an Admin address for ADMIN ONLY tx. There will be 3 days delay on every operational tx except when calling `Withdraw()` function, TimeLock contract is not able to call this function.
 
-### TimeLock Contract Account Permission
+### TimeLock Contract Operations
```
Comment (cuongquangnam) [inline README.md:40]: should be two hash symbol?

File: README.md [hunk 22]
```diff
@@ -51,17 +51,17 @@ To execute only Admin transactions, the developer will need to follow these step
 Salt can be `0x0000000000000000000000000000000000000000000000000000000000000000` for 1st transaction,
 `0x0000000000000000000000000000000000000000000000000000000000000001` for 2nd transaction and so on.
 
-The reason behind choosing different salt is to avoid having the same transaction again.
+The reason behind choosing different salt is to avoid having he same operation identifier again.
```
Comment (cuongquangnam) [inline README.md:54]: "the same operation identifier"

File: README.md [hunk 23]
```diff
@@ -51,17 +51,17 @@ To execute only Admin transactions, the developer will need to follow these step
...
 
-## Operational Transactions
+## Bridge Contract operations
 
 ### Bridge Contract Account Permission
 
 There are only two roles: DEFAULT_ADMIN_ROLE and WITHDRAW_ROLE.
 
-The TimeLock contract is assigned the DEFAULT_ADMIN_ROLE, while another Gnosis Safe is assigned the WITHDRAW_ROLE. The DEFAULT_ADMIN_ROLE has the ability to grant both roles to other addresses, but this change will only take effect after three days.
+The TimeLock contract is assigned the DEFAULT_ADMIN_ROLE, while another Gnosis Safe is assigned the WITHDRAW_ROLE. The DEFAULT_ADMIN_ROLE has the ability to grant both roles to other addresses, but this changes will happen instantly once executed.
```
Comment (cuongquangnam) [inline README.md:64]: "these changes will happen ..."

**Review by cuongquangnam — CHANGES_REQUESTED** [review]
Except for the changes I suggested, everything else LGTM

**Review by cuongquangnam — APPROVED** [review]

File: README.md [hunk 24]
```diff
@@ -31,58 +31,92 @@ TODO
 
 TODO
 
-## Operational Transactions
+####
```
Comment (weiyuan95) [inline README.md:34]:
```suggestion
## Contract Operations
```

**Review by weiyuan95 — APPROVED** [review]

**Pull request merged**